lms/djangoapps/completion/tests/test_models.py

```diff
@@ -142,3 +144,146 @@ def test_cannot_call_submit_completion(self):
                 completion=0.9,
             )
         self.assertEqual(models.BlockCompletion.objects.count(), 1)
+
+
+@ddt.ddt
+class AggregateCompletionTestCase(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(AggregateCompletionTestCase, cls).setUpClass()
+        cls.overrider = waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, True)
+        cls.overrider.__enter__()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.overrider.__exit__(None, None, None)
+        super(AggregateCompletionTestCase, cls).tearDownClass()
+
+    def setUp(self):
+        super(AggregateCompletionTestCase, self).setUp()
+        self.user = UserFactory()
+
+    @ddt.data(
+        # Valid arguments
+        (u'block-v1:edx+test+run+type@video+block@doggos', 'course', 0.5, 1, 0.5),
+    )
+    @ddt.unpack
+    def test_submit_completion_with_valid_data(self, block_key, aggregate_name, earned, possible, expected_percent):
+        block_key_obj = UsageKey.from_string(block_key)
+        obj, is_new = models.AggregateCompletion.objects.submit_completion(
+            self.user,
+            block_key_obj.course_key,
+            block_key_obj,
+            aggregate_name,
+            earned,
+            possible
+        )
+        self.assertTrue(is_new)
+        self.assertEqual(len(models.AggregateCompletion.objects.all()), 1)
+        self.assertEqual(obj.earned, earned)
+        self.assertEqual(obj.possible, possible)
+        self.assertEqual(obj.percent, expected_percent)
+
+    @ddt.data(
+        # Earned greater than possible
+        (u'block-v1:edx+test+run+type@video+block@doggos', 'course', 1.1, 1,
```

Comment:
nit: could you pull this out into a variable on your test class to make it clearer that you're always testing with the same block key?